### ISSUE TITLE
Dirt Rally is still not working

### ISSUE THREAD

**Issue by cybersteel8** [LibreVR/Revive#966]
So, this game has been plaguing Revive for a very long time. I think you already know what I'm going to say - Dirt Rally refuses to open up in VR, only on my desktop, despite all my efforts.

- I am running Windows 7 SP1 x64
- I have tried with Revive 1.1.11 and the current latest version 1.3.0 (I uninstalled first then ran the other installer).
- I have tried with Steam VR BuildID 2704312 and beta BuildID 2707745 (1524242469 is latest patch notes)
- I have executed the DirectX Runtime installer mentioned in #606.
- I have unticked "Use Desktop Game Theatre while SteamVR is active" in Dirt Rally's properties.
- I have the Oculus software installed (and logged in) and tried having it running during some attempts.
- I have clicked Patch and navigated to drt.exe multiple times, and some attempts I've even tried restarting Steam after patching.
- I have opened Documents\My Games\DiRT Rally\hardwaresettings\hardware_settings_config.xml and made sure <oculus enabled="true" ...>

...but no luck :(

Is there anything I have not yet tried? Is there any steps I messed up? Is there any more information I can provide to finally put this problem to rest? I understand many people in the past have had it working after trying one of the aforementioned steps. I believe I have tried everything short of buying it on Oculus.

**Comment by cybersteel8** [comment]
> the patch method for SteamVR games is not very reliable
Could you explain what the Patch option does? I notice that it copies/renames 3 DLLs, but I'm not entirely sure which ones. I see that openvr_api.dll is copied into Dirt Rally's directory, but where do LibOVRProxy32 and LibRevive32 go? I suspect they overwrite xinput1_3.dll and xinput9_1_0.dll based on the "Do you want to overwrite?" prompts, but it doesn't appear to do that correctly (file size and date modified timestamps do not match).

>Revive is only tested on Windows 10
I am happy to help you test + resolve this Windows 7 issue for the sake of backwards compatibility for stubborn folk like me :P

**Comment by Armada651** [comment]
> I suspect they overwrite xinput1_3.dll and xinput9_1_0.dll based on the "Do you want to overwrite?" prompts, but it doesn't appear to do that correctly (file size and date modified timestamps do not match).

Huh, does it match the 64-bit ones?

**Comment by cybersteel8** [comment]
>(file size and date modified timestamps do not match)

I must correct my previous statement. I just took another look, and I realise that I was wrong.
It appears that openvr_api.dll is copying correctly from the x86 folder to the Dirt Rally folder.
It also appears that LibRevive32_1.dll is replacing both xinput1_3.dll and xinput9_1_0.dll
There isn't a dll with the file size matching LibOVRProxy32_1.dll
I know this from inspecting the file size and date modified fields.

**Comment by luzat** [comment]
Did it work on 1.3 for you? I am still on 1.1.11.1 because of Dirt Rally, but would love to upgrade.

**Comment by luzat** [comment]
Previously didn't get it to work for some reason, but confirmed with 1.4.2.0 now that it works fine with the patch method.

**Comment by vorrin** [comment]
I am having this weird issue with that version (1.4.2), I patched Dirt Rally with it some days ago, and it worked fine, I then removed the files to play it non-vr again, and now, after all sorts of copying the files back/re-patching it (which copies 2 xinput and one opnvr_api dlls in the game's folder), it just won't play ball. What seems to happen is that steam does not considedr it a VR game at all, which previously, when this worked, it did (ie. would automatically start steam-vr when starting Dirt Rally). 
I've been trying all googlable suggestions/everything that's outlined here for the whole morning, would anyone have any ideas?

**Comment by cybersteel8** [comment]
I'd like to confirm that I had trouble with the latest version (1.6.2 at this point in time) so I bumped Revive back to 1.4.2.0 and Dirt Rally loaded up in VR immediately. Ensure y'all restart SteamVR when you change your Revive version. I did have the "Dirt Rally doesn't support VR" message from Steam, but it still opened in VR anyway. Check the hardware config file I mentioned in the OP for the <oculus enabled="true"> value.
